cmd/root.go

```diff
@@ -0,0 +1,201 @@
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+)
+
+
+var httpClient    http.Client
+
+var cfgFile string
+
+// InitPayload holds a Vault init request.
+type InitPayload struct {
+	RecoveryShares    int `json:"recovery_shares"`
+	RecoveryThreshold int `json:"recovery_threshold"`
+	SecretShares      int `json:"secret_shares"`
+	SecretThreshold   int `json:"secret_threshold"`
+}
+
+// InitResponse holds a Vault init response.
+type InitResponse struct {
+	Keys               []string `json:"keys"`
+	KeysBase64         []string `json:"keys_base64"`
+	RootToken          string   `json:"root_token"`
+	RecoveryKeys       []string `json:"recovery_keys"`
+	RecoveryKeysBase64 []string `json:"recovery_keys_base64"`
+}
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "vault-init",
+	Short: "Intitialise a specified instance of Vault",
+	Long: `This command makes a request to an instance of Vault at the specified Vault address
+to initialise the instance. This command currently assumes that auto-unseal has
+been setup to occur during Vault initialisation.
+
+Once it has received the token in the response from Vault, it will encrypt and
+store this token on S3 from where it can be used for authentication by entities that
+need to read from or write to the Vault instance.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// make vault init request and get root token
+		fmt.Println("Initialising Vault...")
+		rootToken := initVault()
+		fmt.Println("Initialisation complete")
+
+		// AWS setup
+		region := os.Getenv("AWS_REGION")
+		sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(region)}))
+		var kmsClient = kms.New(sess, aws.NewConfig().WithRegion(region))
+		uploader := s3manager.NewUploader(sess)
+
+		// encrypt tokens with AWS KMS
+		fmt.Println("Encrypting root token...")
+		encryptedToken, errE := kmsClient.Encrypt(&kms.EncryptInput{
+			KeyId: aws.String(fullKeyID(os.Args[1], os.Args[2], region)),
+			Plaintext: []byte(rootToken),
+		})
+		checkError(errE)
+		fmt.Println("Encryption complete.")
+
+		hostname, errH := os.Hostname()
+		if errH != nil {
+			panic(errH)
+		}
+
+		tokenFileName := hostname+ "_token"
+
+		writeToFile(tokenFileName, encryptedToken.CiphertextBlob)
+
+		// upload keys to S3
+		fmt.Println("Uploading encrypted token to S3...")
+		f := openFile(tokenFileName)
+
+		s3Result, errS3 := uploader.Upload(&s3manager.UploadInput{
+			Bucket: aws.String("encrypted-tokens"),
+			Key:    aws.String(tokenFileName),
+			Body:   f,
+		})
+		checkError(errS3)
+		fmt.Println("Encrypted token successfully uploaded to S3 at", s3Result.Location)
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+
```

Comment:
I suspect cobra needs this. Just a suggestion ... could we change this to `func init() {}`?